src/ansys/fluent/core/utils/search.py

```diff
@@ -212,10 +238,320 @@ def inner(tree, path, root_path):
+    """
+    close_matches = closest_allowed_names(word, names)
+    valid_close_matches = [
+        close_match for close_match in close_matches if close_match in names
+    ]
+    return valid_close_matches
+
+
+def _search_whole_word(search_string: str, match_case: bool):
+    """Perform exact search for a word through the Fluent's object hierarchy.
+
+    Parameters
+    ----------
+    search_string: str
+        The word to search for. Semantic search is default.
+    match_case: bool
+        Whether to match case. If ``True`` will match case-insensitive case.
+
+    Returns
+    -------
+        List of search string matches.
+    """
+    names = _get_api_object_names()
+    if match_case:
+        search_string = search_string + "*"
+        _search_wildcard(search_string, names)
+    else:
+        queries = _get_exact_match_for_word_from_names(search_string, names)
+        if queries:
+            _print_search_results(queries, wildcard=False)
+
+
+def _download_nltk_data():
+    """Download NLTK data on-demand."""
+    import nltk
+
+    packages = ["wordnet", "omw-1.4"]
+    for package in packages:
+        nltk.download(package, quiet=True)
+
+
+def _search_semantic(search_string: str, language: str):
+    """Perform semantic search for a word through the Fluent's object hierarchy.
+
+    Parameters
+    ----------
+    search_string: str
+        The word to search for. Semantic search is default.
+    language: str
+        The language for the semantic search.
+        English is default for the semantic search.
+        ISO 639-3 code of the language to be used for semantic search.
+        See `https://omwn.org/omw1.html` for the list of supported languages.
+        The default value is `eng` for English language.
+
+    Returns
+    -------
+    queries: list
+        List of search string matches.
+    """
+    names = _get_api_object_names()
+    similar_keys = set()
+    search_string_synsets = (
+        wn.synsets(search_string.decode("utf-8"), lang=language)
+        if sys.version_info[0] < 3
+        else wn.synsets(search_string, lang=language)
+    )
+    for name in names:
+        api_object_name_synsets = (
+            wn.synsets(name.decode("utf-8"), lang=language)
+            if sys.version_info[0] < 3
+            else wn.synsets(name, lang="eng")
+        )
+        for search_string_synset in search_string_synsets:
+            for api_object_name_synset in api_object_name_synsets:
+                search_string_synset_name = search_string_synset.name().split(".")[0]
+                api_object_synset_name = api_object_name_synset.name().split(".")[0]
+                if (
+                    search_string in api_object_synset_name
+                    or search_string_synset_name in api_object_synset_name
+                ):
+                    similar_keys.add(api_object_synset_name + "*")
+    if similar_keys:
+        for key in similar_keys:
+            _search_wildcard(key)
```

Comment:
Why `_search_semantic` has to call `_search_wildcard`? Shouldn't they be independent?

Reply:
`match_whole_word` and `match_case` are covered in `_search_wildcard` results. Therefore we are calling it for `_search_semantic`.